_playwright-tests/UI/IntrospectRepo.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { closePopupsIfExist, getRowByName } from './helpers/helpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+
+test.describe('Introspect Repositories', () => {
+  const repoName = 'introspection-test';
+  const repoUrl = 'https://fedorapeople.org/groups/katello/fakerepos/zoo/';
+
+  test.beforeEach(async ({ page }) => {
+    await test.step('Navigate to the repository page', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+    });
+  });
+
+  test('Create an introspection repository', async ({ page }) => {
+    await test.step('Cleanup repository, if using the same url', async () => {
+      await deleteAllRepos(page, `&url=${repoUrl}`);
+    });
+
+    await test.step('Open the add repository modal', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
```

Comment:
Why does this need the `first()` selector?

Reply:
if the empty state on the repositories page would be displayed, then there would be two buttons with "Add repositories" and without the `first()` the test would fail on violating the strict locator rule 😅

Reply:
Huh, why are there two button with the same text? :D

Reply:
well there is the one at the top which is always visible and one specifically in the empty state below some help text, we have this way on both repos and templates pages (proposed by UX, at least it's in the mocks), so the `first()` has to be here 😄